Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,488 @@
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Verifies if array is valid.
+        /// </summary>
+        private static bool ArrayIsValid(string[] inputArray)
+        {
+            return inputArray != null && inputArray.Length != 0;
+        }
+
+        /// <summary>
+        /// Calculates the loss for a single person in array.
+        /// </summary>
+        private static decimal CalculateLossForSinglePerson(string personAndBalance)
+        {
+            var balanceList = ConvertStringToDecimalArray(personAndBalance);
+            try
+            {
+                decimal previousBallance = balanceList[^2];
+                decimal currentBalance = balanceList[^1];
+                var loss = currentBalance - previousBallance;
+
+                return loss;
+            }
+            catch (IndexOutOfRangeException)
+            {
+                // Console.WriteLine(ex.Message);
+                // commented it out to keep clean screen
+            }
+            return default;
+        }
+
+        /// <summary>
+        /// Tries to convert a string to decimal.
+        /// </summary>
+        private static decimal ConvertStringToDecimal(string input)
         {
-            return "";
+            decimal.TryParse(input, NumberStyles.Currency, numberFormatInfo, out decimal value);
+
+            return value;
+        }
+
+        /// <summary>
+        /// Converts the numbers split by comma part of the string to an array.
+        /// </summary>
+        private static decimal[] ConvertStringToDecimalArray(string personAndBalance)
+        {
+            if (!InputStringIsValid(personAndBalance))
+            {
+                return default;
+            }
+
+            var array = personAndBalance.Split(',');
+            var newArray = new decimal[array.Length - 1];
+
+            for (int i = 1; i < array.Length; i++)
+            {
+                newArray[i - 1] = ConvertStringToDecimal(array[i]);
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Finds the current balance of a single person in array.
+        /// </summary>
+        private static decimal CurrentBalanceForSinglePerson(string personAndBalance)
+        {
+            var balanceList = ConvertStringToDecimalArray(personAndBalance);
+            try
+            {
+                decimal currentBalance = balanceList[^1];
+
+                return currentBalance;
+            }
+            catch (IndexOutOfRangeException)
+            {
+                // Console.WriteLine(ex.Message);
+                // commented it out to keep clean screen
+            }
+            return default;
+        }
+
+        /// <summary>
+        /// Return index of Max value in decimal array.
+        /// </summary>
+        private static int FindDecimalArrayMax(decimal[] inputArray)
+        {
+            //var max = decimal.Zero;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4971928)